src/modules/flow/mqtt/mqtt.c

```diff
@@ -0,0 +1,452 @@
+{
+    struct client_data *mdata = data;
+    int r;
+    bool in_value;
+
+    r = sol_flow_packet_get_boolean(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mdata->clean_session = in_value;
+
+    return 0;
+}
+
+static int
+keepalive_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct client_data *mdata = data;
+    int r;
+    struct sol_irange in_value;
+
+    r = sol_flow_packet_get_irange(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mdata->keepalive = in_value.val;
+
+    return 0;
+}
+
+static int
+host_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct client_data *mdata = data;
+    int r;
+    const char *in_value;
+
+    r = sol_flow_packet_get_string(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    free(mdata->host);
+    mdata->host = strdup(in_value);
+    SOL_NULL_CHECK(mdata->host, -ENOMEM);
+
+    return 0;
+}
+
+static int
+publish_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct client_data *mdata = data;
+
+    if (!mdata->mqtt) {
+        mdata->pending_publish = true;
+        mqtt_init(mdata);
+    } else {
+        publish(mdata);
+    }
+
+    return 0;
+}
+
+static int
+subscribe_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct client_data *mdata = data;
+
+    if (!mdata->mqtt) {
+        mdata->pending_subscribe = true;
+        mqtt_init(mdata);
+    } else {
+        subscribe(mdata);
+    }
+
+    return 0;
+}
+
+static int
+user_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct client_data *mdata = data;
+    int r;
+    const char *in_value;
+
+    r = sol_flow_packet_get_string(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    free(mdata->user);
+    mdata->user = strdup(in_value);
+    SOL_NULL_CHECK(mdata->user, -ENOMEM);
+
+    return 0;
+}
+
+static int
+qos_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct client_data *mdata = data;
+    int r;
+    struct sol_irange in_value;
+
+    r = sol_flow_packet_get_irange(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    mdata->qos = (sol_mqtt_qos)in_value.val;
+
+    return 0;
+}
+
+static int
+data_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct client_data *mdata = data;
+    int r;
+    struct sol_blob *in_value;
+
+    r = sol_flow_packet_get_blob(packet, &in_value);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (mdata->payload)
+        sol_blob_unref(mdata->payload);
+
+    sol_blob_ref(in_value);
```

Comment:
shouldn't be an issue here, but it's always better to do ref of the new value before doing the unref of the old one. In some cases you may get the same handle and it may be at the last reference, thus doing the unref before will free the handle, that you later ref.